src/BenchmarkDotNet/Engines/GcStats.cs

```diff
@@ -15,8 +15,10 @@ public struct GcStats : IEquatable<GcStats>
         public static readonly long AllocationQuantum = CalculateAllocationQuantumSize();
 
 #if !NET6_0_OR_GREATER
-        private static readonly Func<long> GetAllocatedBytesForCurrentThreadDelegate = CreateGetAllocatedBytesForCurrentThreadDelegate();
+        // do not reorder these, CheckMonitoringTotalAllocatedMemorySize relies on GetTotalAllocatedBytesDelegate being initialized first
         private static readonly Func<bool, long> GetTotalAllocatedBytesDelegate = CreateGetTotalAllocatedBytesDelegate();
+        private static readonly Func<long> GetAllocatedBytesForCurrentThreadDelegate = CreateGetAllocatedBytesForCurrentThreadDelegate();
+        private static readonly bool CanUseMonitoringTotalAllocatedMemorySize = CheckMonitoringTotalAllocatedMemorySize();
```

Comment:
This should be lazy evaluated so that monitoring is not enabled before benchmarks are ran.

Reply:
Would `Lazy` be okay here or maybe a 3 value enum should be used for perf?

Reply:
`Lazy` class is not okay due to its own allocations. I would either use a `bool?`, or move it to its own class (since it's only used in 1 place, the lazy nature of statics is ok).